### PR TITLE
[INTG-996] add: pairing demo

### PR DESCRIPTION
**Pull request by MicheilCathcart**
https://mx51.atlassian.net/browse/INTG-996?atlOrigin=eyJpIjoiY2UxMmQ4NTY4NTg2NDNlNmE2OGFkMTFlYTZhMmUzOGUiLCJwIjoiaiJ9

## Notes

Just for internal testing so no tests, basic ui etc. Will not merge this into main. If wanting to test hit me up for an API key to replace and also I can provide a MXA build with spi cloud pairing functionality.

### CODE REVIEW

File: src/components/SpiCloudPairingNewPage/SpiCloudPairingNewPage.tsx [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    setPairSuccess(false);
+  };
+
+  const handlePairing = async () => {
+    const pairingUrl = 'https://spi-cloud-pairing-api-dev.integration-nonprod.mspenv.io/api/v1/progress-pairing';
+
+    const ApiKey = 'replace-me';
+
+    try {
+      const response = await fetch(pairingUrl, {
```
Comment (MicheilCathcart) [inline src/components/SpiCloudPairingNewPage/SpiCloudPairingNewPage.tsx:48]: @alistair-mx51 here is where the pairing request is being made

**Review by rossi-mx51 — CHANGES_REQUESTED** [review]
Nice work man. 

Just a thing and a potential issue with cancel pairing.

File: src/components/SpiCloudPairingNewPage/SpiCloudPairingNewPage.tsx [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            <Typography className={classes.defaultMargin} variant="body1">
+              Confirmation Code: <strong>{pairingResponse?.data.confirmation_code}</strong>
+            </Typography>
+            <Button color="primary" data-test-id="pairBtn" onClick={() => handleConfirmation()} variant="contained">
+              Confirm
+            </Button>
+            <Button
+              className={classes.cancelPairingBtn}
+              data-test-id="cancelPairBtn"
+              onClick={() => onCancel}
```
[the remark this answers is not in the document]
Comment (MicheilCathcart) [reply src/components/SpiCloudPairingNewPage/SpiCloudPairingNewPage.tsx:141]: Thanks mate

File: src/components/SpiCloudPairingNewPage/SpiCloudPairingNewPage.tsx [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    <Layout>
+      <Container maxWidth="md" className={classes.root}>
+        <Button className={classes.backLink} onClick={() => goToSpiCloudPairing()}>
+          <svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" viewBox="0 0 24 24" fill="none">
+            <path d="M20 11H7.83L13.42 5.41L12 4L4 12L12 20L13.41 18.59L7.83 13H20V11Z" fill="#393F73" />
+          </svg>
+          <span className={classes.backLinkText}>Back to SPI Cloud Pairing</span>
+        </Button>
+
+        {!pairSuccess && (
```
[the remark this answers is not in the document]
Comment (MicheilCathcart) [reply src/components/SpiCloudPairingNewPage/SpiCloudPairingNewPage.tsx:98]: Cool looks tidy

File: src/redux/reducers/SpiCloudSettingsSlice/spiCloudSettingsSlice.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { ISpiCloudSettingsProps } from './interfaces';
+
+const spiCloudSettingsSlice = createSlice({
+  name: 'spiCloudSettings',
+  initialState: {
```
Comment (rossi-mx51) [inline src/redux/reducers/SpiCloudSettingsSlice/spiCloudSettingsSlice.ts:6]: Maybe abstract this initialState out to avoid some duplication ?

**Review by rossi-mx51 — APPROVED** [review]
Looks good, thanks mate. 

A smallish suggestion, but it's not important. Approving now, we can take a look at that in the next PR where we add more envs.

**Pull request merged**